src/report/user-report.controller.ts

```diff
@@ -0,0 +1,51 @@
+import { Request, Response } from 'express';
+import { validate } from 'class-validator';
+import { ReportDto, UpdateStatusDto, ReportService } from '.';
+import { Controller } from '../common/controller.class';
+import { instanceToPlain, plainToInstance } from 'class-transformer';
+import { StatusCodes } from 'http-status-codes';
+
+export class ReportController extends Controller {
+    constructor() {
+        super('/report');
+
+        const router = this.getRouter();
+
+        router.post('/report/:id', this.createReport);
+        router.get('/report', this.getAllReports);
+        router.put('report/:id', this.updateReportStatus);
+    }
+    async getAllReports(req: Request, res: Response) {
+        return res
+            .status(StatusCodes.OK)
+            .send(await ReportService.getAllReports());
+    }
+    async createReport(req: Request, res: Response) {
+        const body = plainToInstance(ReportDto, req.body as ReportDto);
+        body.reason = req.body.reason;
```

Comment:
not required. plainToInstance is doing it for you

~~`body.reason = req.body.reason;`~~